components/datadog/agent/install.go

```diff
@@ -49,16 +49,17 @@ func NewInstaller(vm vm.VM, options ...agentparams.Option) (*Installer, error) {
 		return nil, err
 	}
 
-	var updateConfigTrigger pulumi.StringInput
-	lastCommand, updateConfigTrigger, err = updateAgentConfig(
-		vm.GetFileManager(),
-		env,
-		params.AgentConfig,
-		params.ExtraAgentConfig,
-		os,
-		lastCommand)
-	if err != nil {
-		return nil, err
+	fileChangeTriggers := make(map[string]pulumi.StringInput)
+	var trigger pulumi.StringInput
+	for _, input := range []struct{ path, content string }{
+		{"datadog.yaml", params.AgentConfig},
+		{"system-probe.yaml", params.SystemProbeConfig},
+		{"security-agent.yaml", params.SecurityAgentConfig}} {
+		lastCommand, trigger, err = updateConfig(vm.GetFileManager(), input.path, input.content, params.ExtraAgentConfig, env, os, lastCommand)
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Nice !